app/models/forms/attribute_reset_collection.rb

```diff
@@ -0,0 +1,23 @@
+module Forms
+  class AttributeResetCollection
+    ResetData = Struct.new(:attributes_to_reset, :master_attribute, :enabled_value)
+
+    def initialize
+      @collection = []
+    end
+
+    def add(*args)
+      @collection << ResetData.new(*args)
+    end
+
+    def perform(form)
+      @collection.each do |reset_obj|
+        next if form.public_send(reset_obj.master_attribute) == reset_obj.enabled_value
+
+        reset_obj.attributes_to_reset.each do |attribute|
+          form.public_send("#{attribute}=", nil)
```

Comment:
I did consider that, but didn't go anywhere with it when I realised we prescribe choices for some of the fields already(e.g. arson & csra) & this just felt like an extension of that. It kind of feels like theres either a small layer missing or the form objects are a little beyond what you'd expect from a form object.